gulpfile.js

```diff
@@ -416,3 +418,46 @@ gulp.task('test-e2e', function() {
     'stop-webserver'
   );
 });
+
+/**
+ * Deploys production-optimized app build to the "origin/gh-branch". Every new
+ * deployment comes with a separate commit attributed with the date and source
+ * branch revision. If tree has some changes that are not committed yet, there
+ * will be a warning, but all local changes will be deployed anyway.
+ */
+gulp.task('deploy', ['build-production'], function() {
+  const wrapIntoPromise = (method, args) => new Promise((resolve, reject) => {
+    git[method].call(
+      git, args, (err, result) => err ? reject(err) : resolve(result)
+    );
+  });
+
+  return Promise
+    .all([
+      wrapIntoPromise('revParse', { args: '--short HEAD', quiet: true }),
+      wrapIntoPromise('status', { args: '--porcelain', quiet: true }),
+    ])
+    .then((results) => {
+      const revision = results[0];
+      const status = results[1];
+      const deployDate = new Date().toUTCString();
+
+      let message = `Deployed on '${deployDate}', based on '${revision}'`;
+
+      if (status) {
+        console.log(
+          '\x1b[31m',
+          `You have uncommitted changes that will be deployed!\n${status}`,
+          '\x1b[0m'
+        );
+
+        message += ' (includes uncommitted changes)';
+      }
+
+      return new Promise((resolve, reject) => {
+        merge(gulp.src(`${DIST_APP_ROOT}**/*`).pipe(ghPages({ message })))
```

Comment:
Ah, forgot to mention - this `merge` around stream is workaround I was able to find so far, the issue is that `.on('end', ...` without `merge` wrapper is called at some weird moment and plugin doesn't have time to finish the job - it's definitely bug somewhere - just need more time to investigate.